src/View/Sidebar/BookmarkListBox.vala

```diff
@@ -195,12 +195,11 @@ public class Sidebar.BookmarkListBox : Gtk.Box, Sidebar.SidebarListInterface {
                                        int pos = 0) {
 
         int pinned = 0; // Assume pinned items only at start and end of list
-        foreach (unowned var child in list_box.get_children ()) {
-            if (((BookmarkRow)child).pinned) {
-                pinned++;
-            } else {
-                break;
-            }
+        int index = 0;
+        Gtk.ListBoxRow? row = list_box.get_row_at_index (index++);
```

Comment:
You start the index at 0 but then immediately make it 1 here. That seems a bit overcomplicated? Why not just start the index where you need it to start? :sweat_smile:

Reply:
Unless I'm mistaken, when `++` is post-fixed it operates *after* the variable is used so the current code is equivalent to 
```
Gtk.ListBoxRow? row = list_box.get_row_at_index (index);
index = index + 1;
```

(when the operator is pre-fixed it increments the variable *before* using it)

I'll double-check though and see if there is a clearer, equally concise way.